tests/type_inference/test_literals.py

```diff
@@ -146,6 +146,27 @@ def test_single_assign(variable, value):
     assert target_value.type_constraints.type == target_type
 
 
+@given(cs.random_dict_variable_value(min_size=1))
+def test_multi_target_assign(variables_dict):
+    """Test multi-target assignment statements; verify unification of type variables."""
+    program = ""
+    for variable_name in variables_dict:
+        assume(not iskeyword(variable_name))
+        program += variable_name + ", "
+    program += " = "
+    for variable_name in variables_dict:
+        program += repr(variables_dict[variable_name]) + ", "
+    module = _parse_text(program)
+    # Assign node for this type of expr is the first node in the body of the module
+    assign_nodes = [node for node in module.nodes_of_class(astroid.Assign)]
+    targets_list = assign_nodes[0].targets[0].elts
+    for i in range(len(targets_list)-1):
+        target_node = targets_list[i]
+        target_type_var = target_node.frame().type_environment.lookup_in_env(target_node.name)
+        target_type = TYPE_CONSTRAINTS.lookup_concrete(target_type_var)
+        assert type(variables_dict[target_node.name]) == target_type
```

Comment:
This is a good start, but can be improved. The right-side of the Assign node should be a Tuple. What you should really do is:

1. `zip` the `targets_list` and the elements of the RHS tuple.
2. For each pair in the `zip`, check whether the `lookup_concrete` on the identifier matches the `type_constraint.type` of the corresponding RHS tuple element.